### PR TITLE
feat: Move the placeOrder() logic from the checkout library to the order library

### LINKED ISSUE #14649: Move the placeOrder() logic from the checkout library to the order library
### What to do
We should move the logic of "placing the order" from the `checkout` lib to the `order` lib. This includes  the adapter, controller and the facade (and maybe something else too...).

### Reasons
During the cart lib's code splitting refactor, we discovered that `occCheckoutAdapter.placeOrder()` (in the lazy-loaded `checkout` lib) uses the `ORDER_NORMALIZER` token, which is provided only in the lazy loaded `order` lib. This unclear lib cross-dependendy causes a few problems:

- Now the action of "placing the order" in the last step of the checkout won't work if we don't lead the `order` lib "artificially" in advance.
- it feels natural to me to have the "placing the order" action in the Order domain lib. Although it's debatable whether it should belong to the checkout or the order domain, there are no known disadvantages of having it in the Order domain. But there are problems of having it in the Checkout domain.
- Having the "place order" in the core facade of the Order domain feels natural also from the OCC API point of view. The OCC endpoint for placing the order is a part of the [OCC Orders module](https://40.76.109.9:9002/occ/v2/swagger-ui.html#/Orders/placeOrder): `[POST] /{baseSiteId}/users/{userId}/order`.

### When to do
We should consider doing this ticket **ONLY AFTER** the ongoing refactors of the cart lib and the checkout lib are done (merged to develop in a releasable state). It's to to avoid unnecessary conflicts that could complicate and delay the ongoing cart and checkout lib refactors.

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-14649

### CODE REVIEW

File: feature-libs/order/components/unnamed-order-confirmation/unnamed-order-confirmation-thank-you-message/unnamed-order-confirmation-thank-you-message.component.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div class="cx-page-header">
     <span class="cx-page-title">
       {{ 'checkoutOrderConfirmation.confirmationOfOrder' | cxTranslate }}
-      {{ order.code }}
+      {{ order.replenishmentOrderCode ?? order.code }}
```
Comment (znikola) [inline feature-libs/order/components/unnamed-order-confirmation/unnamed-order-confirmation-thank-you-message/unnamed-order-confirmation-thank-you-message.component.html]: This is a very nice idea.
We should consider how to do it once we are able to separate the repl from the rest of order code.

File: feature-libs/checkout/base/occ/config/default-occ-checkout-config.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const defaultOccCheckoutConfig: OccConfig = {
       endpoints: {
         setDeliveryAddress:
           'users/${userId}/carts/${cartId}/addresses/delivery',
-        placeOrder: 'users/${userId}/orders?fields=FULL',
+        // placeOrder: 'users/${userId}/orders?fields=FULL',
```
Comment (znikola) [inline feature-libs/checkout/base/occ/config/default-occ-checkout-config.ts]: Let's mark it with a todo comment, e.g.: `TODO:#move-order`

File: feature-libs/checkout/base/root/checkout-root.module.ts [hunk 3]
```diff
@@ -42,7 +37,7 @@ export function defaultCheckoutComponentsConfig() {
       [CHECKOUT_FEATURE]: {
         cmsComponents: CHECKOUT_BASE_CMS_COMPONENTS,
         // TODO:#checkout - remove ORDER_FEATURE once we move the order placing functionality to the order lib
```
Comment (znikola) [inline feature-libs/checkout/base/root/checkout-root.module.ts]: Let's not forget to remove `TODO:#`

File: feature-libs/checkout/base/root/config/default-checkout-routing-config.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ export const defaultCheckoutRoutingConfig: RoutingConfig = {
       checkoutDeliveryMode: { paths: ['checkout/delivery-mode'] },
       checkoutPaymentDetails: { paths: ['checkout/payment-details'] },
       checkoutReviewOrder: { paths: ['checkout/review-order'] },
-      orderConfirmation: { paths: ['order-confirmation'] },
+      // orderConfirmation: { paths: ['order-confirmation'] },
```
Comment (znikola) [inline feature-libs/checkout/base/root/config/default-checkout-routing-config.ts]: Let's mark it with a todo comment, e.g.: `TODO:#move-order`

File: feature-libs/checkout/scheduled-replenishment/occ/model/occ-checkout-scheduled-replenishment-endpoints.model.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { OccEndpoint } from '@spartacus/core';
 
 export interface CheckoutScheduledReplenishmentOccEndpoints {
-  scheduleReplenishmentOrder?: string | OccEndpoint;
+  abc?: string | OccEndpoint;
```
Comment (znikola) [inline feature-libs/checkout/scheduled-replenishment/occ/model/occ-checkout-scheduled-replenishment-endpoints.model.ts]: Let's mark it with a todo comment, e.g.: TODO:#move-order

File: feature-libs/checkout/scheduled-replenishment/root/checkout-scheduled-replenishment-root.module.ts [hunk 6]
```diff
@@ -39,7 +30,7 @@ export function defaultCheckoutComponentsConfig() {
 @NgModule({
   imports: [CheckoutB2BRootModule, CheckoutScheduledReplenishmentEventModule],
   providers: [
-    provideDefaultConfig(defaultCheckoutScheduledReplenishmentRoutingConfig),
+    // provideDefaultConfig(defaultCheckoutScheduledReplenishmentRoutingConfig),
```
Comment (znikola) [inline feature-libs/checkout/scheduled-replenishment/root/checkout-scheduled-replenishment-root.module.ts]: So the routing config is going to live in the order lib?

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/checkout/scheduled-replenishment/root/checkout-scheduled-replenishment-root.module.ts (hunk 6 above)
Comment (bgambocjaviniar) [reply feature-libs/checkout/scheduled-replenishment/root/checkout-scheduled-replenishment-root.module.ts]: it's in order lib, I just didn't want both libs have conflict if that will be an issue

File: feature-libs/order/components/guards/order-confirmation.guard.spec.ts [hunk 7]
```diff
@@ -7,7 +7,7 @@ import { OrderConfirmationGuard } from './order-confirmation.guard';
 import createSpy = jasmine.createSpy;
 
 class MockUnnamedService implements Partial<UnnamedFacade> {
-  getCurrentOrderDetails = createSpy().and.returnValue(of(undefined));
+  getOrderDetails = createSpy().and.returnValue(of(undefined));
```
Comment (znikola) [inline feature-libs/order/components/guards/order-confirmation.guard.spec.ts:10]: What do you think about `getPlacedOrderDetails`?

File: feature-libs/order/occ/config/default-occ-order-config.ts [hunk 8]
```diff
@@ -28,6 +28,10 @@ export const defaultOccOrderConfig: OccConfig = {
         replenishmentOrderHistory:
           'users/${userId}/replenishmentOrders?fields=FULL,replenishmentOrders(FULL, purchaseOrderNumber)',
         /** scheduled replenishment endpoints end */
+
+        /** placing or scheduling an order endpoints start **/
```
Comment (znikola) [inline feature-libs/order/occ/config/default-occ-order-config.ts]: The comment is a bit misleading by stating this endpoint is for both placing a regular order _and_ scheduling an order

File: feature-libs/checkout/scheduled-replenishment/components/checkout-place-order/checkout-place-order.component.ts [hunk 9]
```diff
@@ -36,15 +36,15 @@ export class CheckoutScheduledReplenishmentPlaceOrderComponent
...
   constructor(
-    protected checkoutFacade: OrderFacade,
+    protected orderFacade: OrderFacade,
     protected routingService: RoutingService,
     protected fb: FormBuilder,
     protected launchDialogService: LaunchDialogService,
     protected vcr: ViewContainerRef,
     protected checkoutReplenishmentFormService: CheckoutReplenishmentFormService,
-    protected checkoutScheduledReplenishmentFacade: ScheduledReplenishmentFacade
+    protected scheduledReplenishmentFacade: ScheduledReplenishmentOrderFacade
```
Comment (znikola) [inline feature-libs/checkout/scheduled-replenishment/components/checkout-place-order/checkout-place-order.component.ts]:
```suggestion
    protected scheduledReplenishmentOrderFacade: ScheduledReplenishmentOrderFacade
```

File: feature-libs/order/core/facade/order.service.ts [hunk 10]
```diff
@@ -1,205 +1,103 @@
...
-import { CANCEL_ORDER_PROCESS_ID, StateWithOrder } from '../store/order-state';
-import { OrderSelectors } from '../store/selectors/index';
+import { Order, OrderFacade, OrderPlacedEvent } from '@spartacus/order/root';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map, switchMap, take, tap } from 'rxjs/operators';
+import { OrderConnector } from '../connectors/order.connector';
 
 @Injectable()
 export class OrderService implements OrderFacade {
+  protected order$ = new BehaviorSubject<Order | undefined>(undefined);
```
Comment (znikola) [inline feature-libs/order/core/facade/order.service.ts]:
```suggestion
  protected placedOrder$ = new BehaviorSubject<Order | undefined>(undefined);
```
?

File: feature-libs/order/root/facade/order.facade.ts [hunk 11]
```diff
@@ -2,128 +2,39 @@ import { Injectable } from '@angular/core';
...
+      facade: OrderFacade,
+      feature: ORDER_CORE_FEATURE,
+      methods: [
+        'getOrderDetails',
+        'clearPlacedOrder',
+        'setPlacedOrder',
+        'placeOrder',
+      ],
+      // TODO:#deprecation-checkout - remove once we remove ngrx
+      async: true,
```
Comment (znikola) [inline feature-libs/order/root/facade/order.facade.ts]: We don't use store in the facade's impl, and I think we can remove this

File: feature-libs/order/root/facade/scheduled-replenishment.facade.ts [hunk 12]
```diff
@@ -1,22 +1,22 @@
...
   providedIn: 'root',
   useFactory: () =>
     facadeFactory({
-      facade: CheckoutScheduledReplenishmentFacade,
-      feature: CHECKOUT_CORE_FEATURE,
+      facade: ScheduledReplenishmentOrderFacade,
+      feature: ORDER_CORE_FEATURE,
       methods: ['scheduleReplenishmentOrder'],
       // TODO:#deprecation-checkout - remove once we remove ngrx
       async: true,
```
Comment (znikola) [inline feature-libs/order/root/facade/scheduled-replenishment.facade.ts]: We don't use store in the facade's impl, and I think we can remove this

**Review by znikola — CHANGES_REQUESTED** [review]

**Review by znikola — APPROVED** [review]

**Review by plabadie — APPROVED** [review]

**Pull request merged**